### PR TITLE
core: make SMALL_PAGE_MASK and friends of type paddr_t

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Makes SMALL_PAGE_MASK, CORE_MMU_PGDIR_MASK, CORE_MMU_USER_CODE_MASK and
CORE_MMU_USER_PARAM_MASK of type paddr_t to allow correct masking of
significant bits.

Example:
	extern paddr_t addr;
	paddr_t page_addr = addr & ~SMALL_PAGE_MASK

If paddr_t is a 64-bit type SMALL_PAGE_MASK must also be 64-bit wide or
the ~ operation will not set all the higher bits.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: core/arch/arm/include/mm/core_mmu.h [hunk 1]
```diff
@@ -19,8 +19,8 @@
 
 /* A small page is the smallest unit of memory that can be mapped */
 #define SMALL_PAGE_SHIFT	12
-#define SMALL_PAGE_MASK		(SMALL_PAGE_SIZE - 1)
 #define SMALL_PAGE_SIZE		BIT(SMALL_PAGE_SHIFT)
+#define SMALL_PAGE_MASK		((paddr_t)(SMALL_PAGE_SIZE - 1))
```
Comment (jforissier) [inline core/arch/arm/include/mm/core_mmu.h]: I think the inner parentheses are not necessary?
```
((paddr_t)(SMALL_PAGE_SIZE - 1))
          ^                   ^
           \-------------------\-- here
```

Comment (jenswi-linaro) [reply core/arch/arm/include/mm/core_mmu.h]: You're right, I'll fix.

**Comment by jenswi-linaro** [comment]
Update

**Comment by jenswi-linaro** [comment]
@jforissier, you may want to amend that merge.

**Comment by jforissier** [comment]
Oops! Sorry :open_mouth: Master branch reset...

**Comment by jforissier** [comment]
...and commits now squashed & merged.

**Pull request merged**